DSharpPlus.UnifiedCommands/Internals/CommandModuleRegister.cs

```diff
@@ -0,0 +1,344 @@
+                }
+                else
+                {
+                    option = new();
+                    option.WithName(attribute.Name).WithDescription(attribute.Description)
+                        .WithType(ApplicationCommandOptionType.SubCommand);
+                }
+
+                List<ApplicationMethodParameterData> parameters = new();
+                foreach (ParameterInfo parameter in method.GetParameters())
+                {
+                    ApplicationOptionAttribute? name = parameter.GetCustomAttribute<ApplicationOptionAttribute>();
+                    if (name is null)
+                    {
+                        throw new Exception("Parameter needs to have `ApplicationNameAttribute` marked.");
+                    }
+
+                    ApplicationMethodParameterData data = new(name.Name)
+                    {
+                        IsNullable = nullabilityContext.Create(parameter).WriteState is NullabilityState.Nullable
+                    };
+
+                    if (parameter.ParameterType == typeof(string))
+                    {
+                        data.Type = ApplicationCommandOptionType.String;
+                    }
+                    else if (parameter.ParameterType == typeof(long))
+                    {
+                        data.Type = ApplicationCommandOptionType.Integer;
+                    }
+                    else if (parameter.ParameterType == typeof(double))
+                    {
+                        data.Type = ApplicationCommandOptionType.Number;
+                    }
+                    else if (parameter.ParameterType == typeof(bool))
+                    {
+                        data.Type = ApplicationCommandOptionType.Boolean;
+                    }
+                    else if (parameter.ParameterType == typeof(DiscordUser))
+                    {
+                        data.Type = ApplicationCommandOptionType.User;
+                    }
+                    else if (parameter.ParameterType == typeof(DiscordChannel))
+                    {
+                        data.Type = ApplicationCommandOptionType.Channel;
+                    }
+                    else if (parameter.ParameterType == typeof(DiscordRole))
+                    {
+                        data.Type = ApplicationCommandOptionType.Role;
+                    }
+                    else if (parameter.ParameterType.IsSubclassOf(typeof(IMention)) ||
+                             parameter.ParameterType == typeof(IMention))
+                    {
+                        data.Type = ApplicationCommandOptionType.Mentionable;
+                    }
+                    else
+                    {
+                        data.Type = parameter.ParameterType == typeof(DiscordAttachment)
+                            ? ApplicationCommandOptionType.Attachment
+                            : throw new Exception("Not a valid parameter type.");
```

Comment:
Would be nice if the error message contains the methodname, so the user can see easily where they have to change something